### DISCUSSION TITLE
boolector formal engine crashed with broken pipe error

### DISCUSSION THREAD

**Discussion by buttercutter** [YosysHQ/yosys#2950] [Q&A]
## Steps to reproduce the issue

In this [sby file](https://github.com/promach/noc/blob/master/spidergon.sby#L15), comment out `smtbmc yices` and then uncomment `smtbmc boolector` , and also modify `proof: depth 20` to `proof: depth 1000`, then run `sby -f spidergon.sby`

## Expected behavior

The formal engine should finish the BMC and induction verification without crashing and terminated on its own

## Actual behavior

The formal engine crashed and gave me `broken pipe` error

![boolector_broken_pipe_error](https://user-images.githubusercontent.com/3324659/104081816-37bcd580-526c-11eb-8a05-a28bc5d3a673.png)

**Comment by whitequark** [comment]
Are you running out of memory, perhaps?
